### DISCUSSION TITLE
Support for Oomph targlets in Tycho

### DISCUSSION THREAD

**Discussion by eliericha** [eclipse-tycho/tycho#340] [Q&A]
Hello folks,

I maintain a target platform definition as a Targlet in an Oomph setup. My targlet is equipped with an annotation that triggers the generation of a plain .target file containing InstallableUnit locations that Tycho happily consumes.

Recently I have had to add a Maven dependency to my target platform. Oomph targlets do not support Maven repositories so I could not add that to the Oomph setup directly.

Instead, with the [m2e PDE Integration](https://www.modumind.com/2021/01/11/including-maven-artifacts-in-an-eclipse-rcp-target-platform/) I was able to craft a .target file outside the Oomph setup where I combined a location with `type="Maven"` and a location with `type="Targlet"`. This gave me a correct target platform in Eclipse, but not in Tycho because it does not support the Targlet location type.

Has the support for Targlets in Tycho target platform resolution been considered so far? Do we have a sense of the feasibility?

I currently have the workaround of maintaining the .target file with `Maven` and `InstallableUnit` locations. But that format is less efficient than Oomph targlets which have faster resolution and reuse the common plugin pool shared by all Eclipse installations.

And finally, does anyone have other ideas on how I might add a Maven dependency to an Oomph-defined target platform?

Thanks!

**Comment by mickaelistria** [comment]
> Has the support for Targlets in Tycho target platform resolution been considered so far?

Not that I am aware of. But what particularly makes you need to use Targlets? What are you missing in PDE/m2e ?

> But that format is less efficient than Oomph targlets which have faster resolution and reuse the common plugin pool shared by all Eclipse installations.

Have you reported this issue to PDE and Tycho? If there is some "standard" local p2 repo that PDE and Tycho could consider, it could make sense to add it directly in the consumer plugin to not depend on an Oomph-specific location type.

**Comment by laeubi** [comment]
For the record, targlets are described here: https://wiki.eclipse.org/Oomph_Targlets

While it is possible to support targelts you should first consider how some specifics are handeled:

1. How to handle Source Locator
2. targlets support some shared state, so one needs to consider where tycho should store them, should it be possible to survive maven invocations and alike
3. the shared bundle pool is generally a user specific location that won't be accessible for most maven builds
4. How should grouped sites be handled or any other thing that might be controlled by the UI

Beside from that, if you simply want tycho to *resolve* it might not be necessary to add support for all of this. Maybe Ed can give some more insights...

At least you have the following options to give more priority on this this:

- Open an issue and describe the requirements
- Provide a small (executable) reproducer that shows your problem in form of an [itest showing your problem](https://github.com/eclipse/tycho/wiki#providing-an-integration-test).
- Start to provide a patch that fixes the problem
- You can [sponsor me](https://github.com/sponsors/laeubi) (or search for someone else with java skills) and offer a funding for this feature if you need dedicated-support
